src/CMakeLists.txt

```diff
@@ -524,11 +526,15 @@ endif()
 
 add_library(common STATIC ${ceph_common_objs})
 target_link_libraries(common ${ceph_common_deps})
-add_dependencies(common legacy-option-headers ${jaeger_base})
+if(WITH_JAEGER)
+add_dependencies(common legacy-option-headers jaeger_base)
+endif()
```

Comment:
@ideepika  maybe you meant to add the `legacy-option-headers` even if WITH_JAEGER is false?
think that could cause the API failures